schema_version: 1.2.0
id: CVE-2017-2809
aliases: []
modified: 2017-10-02T12:09Z
published: 2017-09-14T19:29Z
details: An exploitable vulnerability exists in the yaml loading functionality of
  ansible-vault before 1.0.5. A specially crafted vault can execute arbitrary python
  commands resulting in command execution. An attacker can insert python into the
  vault to trigger this vulnerability.
severity:
- type: CVSS_V3
  score: 7.8
- type: CVSS_V2
  score: 6.8
affected:
- package:
    name: ansible-vault
    cpe: cpe:2.3:a:ansible-vault_project:ansible-vault
- ranges:
  - type: GIT
    repo: https://github.com/tomoh1r/ansible-vault
    events:
    - introduced: '0'
    - fixed: 3f8f659ef443ab870bb19f95d43543470168ae04
- database_specific:
    CWE:
    - CWE-94
    CVSS:
      Score: 7.8
      Severity: HIGH
      Code: CVSS:3.0/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H
references:
- type: WEB
  url: https://github.com/tomoh1r/ansible-vault/blob/v1.0.5/CHANGES.txt
- type: WEB
  url: http://www.securityfocus.com/bid/100824
- type: WEB
  url: https://www.talosintelligence.com/vulnerability_reports/TALOS-2017-0305
- type: FIX
  url: https://github.com/tomoh1r/ansible-vault/commit/3f8f659ef443ab870bb19f95d43543470168ae04
- type: REPORT
  url: https://github.com/tomoh1r/ansible-vault/issues/4
